lib/http/request/_request_timing.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+var times = {};
+var start = function() {};
+var createResponseCallback = function() {};
+var createSocketCallback = function() {};
+var _ENV = process.env;
+
+var dogStatsDHost = _ENV.CN_DATADOG_HOST || '127.0.0.1';
+var dogStatsDPort = _ENV.CN_DATADOG_PORT || 8120;
+
+if (_ENV.HTTP_REQUEST_TIMING) {
+
+  var DogStatsD = require('hot-shots');
+  var statsClient = new DogStatsD({
+    host: dogStatsDHost,
+    port: dogStatsDPort,
+    maxBufferSize: 512
+  });
+
+  /**
+   * Take a time reading with a specific name, as a delta
+   * from another mark. The second argment is an array of mark names,
+   * so you can specify the time taken to 'firstByte' as as delta
+   * from 'tlsHandshake' for HTTPS connections, or from 'tcpConnection'
+   * for HTTP connections.
+   *
+   * @param {string} name name of the mark to record
+   * @param {array} deltaFrom a list of mark names to use as origin
+   * @param {Array} tags array of tag strings
+   * @returns {undefined} undefined
+   */
+  function timing(name, deltaFrom, tags) {
```

Comment:
I've not used it before, but Node has a Performance Timing API now: https://nodejs.org/api/perf_hooks.html. It only became available in Node 9.x, so it won't work for us. Probably good to at least be aware of.